packages/components/src/button/style.scss

```diff
@@ -287,6 +287,25 @@
 	vertical-align: baseline;
 }
 
+.button.is-transparent {
+	background: transparent;
+	border-color: var( --color-sidebar-text );
+	display: flex;
+    justify-content: center;
+
+	&,
+	&:visited {
+		color: var( --color-sidebar-text );
+	}
+
+	&:hover {
+		border-color: var( --color-sidebar-menu-hover-text );
+		color: var( --color-sidebar-menu-hover-text );
```

Comment:
Let's use different variables here. It's not right to use domain-related variables inside generic components, because these variables won't ever exist if used outside that specific context/domain.

I suggest using already-known variables, or renaming the variables to something that does not include context:
- `--button-text-color` instead of `--color-sidebar-text`;
- `--button-text-color-hover` instead of `--color-sidebar-menu-hover-text).

Then, on usage, set the value of such variables in the immediate parent of the button to `var( --color-sidebar-text )` and `var( --color-sidebar-menu-hover-text )`, like:

```css
.sidebar {
  --button-text-color: var( --color-sidebar-text );
  --button-text-color-hover: var( --color-sidebar-menu-hover-text );
}
```

That's considering the transparent button is a child of the `.sidebar` element.

Reply:
@sejas, iterating upon my previous comment, here's a diff:

```diff
diff --git a/client/components/site-selector/add-site.tsx b/client/components/site-selector/add-site.tsx
index 478e2fe919..d252c57bcd 100644
--- a/client/components/site-selector/add-site.tsx
+++ b/client/components/site-selector/add-site.tsx
@@ -23,6 +23,7 @@ const SiteSelectorAddSite: FunctionComponent = () => {
 			primary
 			href={ `${ onboardingUrl() }?ref=calypso-selector` }
 			onClick={ recordAddNewSite }
+			className="site-selector__add-new-site"
 		>
 			{ translate( 'Add new site' ) }
 		</Button>
diff --git a/client/components/site-selector/style.scss b/client/components/site-selector/style.scss
index a8df707c13..66e09a19a9 100644
--- a/client/components/site-selector/style.scss
+++ b/client/components/site-selector/style.scss
@@ -229,3 +229,7 @@
 .site-selector__no-results + .site-selector__hidden-sites-message {
 	display: none;
 }
+
+.site-selector__actions :not(.site-selector__add-new-site) {
+	--color-accent: var(--color-sidebar-menu-hover-text);
+}
diff --git a/client/my-sites/sidebar/add-new-site.jsx b/client/my-sites/sidebar/add-new-site.jsx
index 6492ddd5be..fd92e9d052 100644
--- a/client/my-sites/sidebar/add-new-site.jsx
+++ b/client/my-sites/sidebar/add-new-site.jsx
@@ -21,7 +21,12 @@ export const AddNewSite = ( { title } ) => {
 
 	return (
 		<li className="sidebar__actions">
-			<Button transparent href={ `${ onboardingUrl() }?ref=calypso-sidebar` } onClick={ onClick }>
+			<Button
+				className="sidebar__add-new-site-button"
+				transparent
+				href={ `${ onboardingUrl() }?ref=calypso-sidebar` }
+				onClick={ onClick }
+			>
 				<span class="sidebar__action--collapsed dashicons dashicons-plus-alt"></span>
 				<span>{ title }</span>
 			</Button>
diff --git a/client/my-sites/sidebar/style.scss b/client/my-sites/sidebar/style.scss
index c0fb324869..a81b80d32c 100644
--- a/client/my-sites/sidebar/style.scss
+++ b/client/my-sites/sidebar/style.scss
@@ -47,7 +47,11 @@ $font-size: rem(14px);
 	align-items: stretch;
 	justify-content: flex-end;
 	margin-top: auto;
-	padding: 8px 8px 0 8px ;
+	padding: 8px 8px 0 8px;
+}
+
+.sidebar__add-new-site-button {
+	--color-accent: var(--color-sidebar-menu-hover-text);
 }
 
 .sidebar__action--collapsed {
diff --git a/packages/components/src/button/index.stories.jsx b/packages/components/src/button/index.stories.jsx
index 886c4821b8..61de6e1565 100644
--- a/packages/components/src/button/index.stories.jsx
+++ b/packages/components/src/button/index.stories.jsx
@@ -31,7 +31,7 @@ export const Transparent = () => {
 	const sidebarStyle = {
 		...wrapperStyle,
 		gap: '1em',
-		backgroundColor: 'var( --color-sidebar-background )',
+		backgroundColor: 'black',
 	};
 	return (
 		<>
diff --git a/packages/components/src/button/stories.scss b/packages/components/src/button/stories.scss
index 3df43bd5e5..d13167a937 100644
--- a/packages/components/src/button/stories.scss
+++ b/packages/components/src/button/stories.scss
@@ -1,5 +1,3 @@
 .theme-default {
-	--color-sidebar-background: #23282d;
-	--color-sidebar-text: #eee;
-	--color-sidebar-menu-hover-text: #00b9eb;
+	color: #fff;
 }
diff --git a/packages/components/src/button/style.scss b/packages/components/src/button/style.scss
index f0a1d19871..82aa15192b 100644
--- a/packages/components/src/button/style.scss
+++ b/packages/components/src/button/style.scss
@@ -289,18 +289,19 @@
 
 .button.is-transparent {
 	background: transparent;
-	border-color: var( --color-sidebar-text );
+	color: inherit;
+	border-color: currentColor;
 	display: flex;
-    justify-content: center;
+	justify-content: center;
 
 	&,
 	&:visited {
-		color: var( --color-sidebar-text );
+		color: inherit;
 	}
 
 	&:hover {
-		border-color: var( --color-sidebar-menu-hover-text );
-		color: var( --color-sidebar-menu-hover-text );
+		border-color: var(--color-accent);
+		color: var(--color-accent);
 		background: transparent;
 	}
 }
```

Reply:
So my suggestion is that we use the current color as the border and text color, and the accent color in the hover state. This feels like the most generic solution possible.

Reply:
@zaguiini 
I've added `--button-text-color` and `--button-text-color-hover` CSS variables. I think these names are more descriptive.
But because these variables don't exist for sure, I've added other variables as a fallback. These variables are from the calypso color scheme: https://github.com/Automattic/wp-calypso/tree/trunk/packages/calypso-color-schemes

- Commit: https://github.com/Automattic/wp-calypso/pull/67322/commits/5c75038f17b7afb44073cd638f2f8b6b8cfe50b1#diff-0688ad494526b3b921c4444dc8bc6f63efc4ad74c7c1123b061490cfc57f293a

```CSS
	&,
	&:visited {
		color: var( --button-text-color, currentcolor );
	}
	&:hover {
		border-color: var( --button-text-color-hover, var( --color-primary-40 ) );
		color: var( --button-text-color-hover, var( --color-primary-40 ) );
	}
```

Reply:
That's fine, too! I would go one step further and rename them to `--text-color` and `--text-color-hover`, dropping the `--button` prefix. And, to top it, apply those variables to the element itself instead of the parent. That's even better than my diff, honestly xD